rust/crd/src/s3logdir.rs

```diff
@@ -0,0 +1,170 @@
+
+#[derive(Snafu, Debug, EnumDiscriminants)]
+#[strum_discriminants(derive(IntoStaticStr))]
+#[allow(clippy::enum_variant_names)]
+pub enum Error {
+    #[snafu(display("s3 bucket error"))]
+    S3Bucket {
+        source: stackable_operator::error::Error,
+    },
+    #[snafu(display("missing bucket name for history logs"))]
+    BucketNameMissing,
+    #[snafu(display("tls non-verification not supported"))]
+    S3TlsNoVerificationNotSupported,
+    #[snafu(display("ca-cert verification not supported"))]
+    S3TlsCaVerificationNotSupported,
+}
+
+pub struct S3LogDir {
+    bucket: InlinedS3BucketSpec,
+    prefix: String,
+}
+
+impl S3LogDir {
+    pub async fn resolve(
+        log_file_dir: Option<&LogFileDirectorySpec>,
+        namespace: Option<String>,
+        client: &stackable_operator::client::Client,
+    ) -> Result<Option<S3LogDir>, Error> {
+        #[allow(irrefutable_let_patterns)]
+        let (s3bucket, prefix) =
+            if let Some(S3(S3LogFileDirectorySpec { bucket, prefix })) = log_file_dir {
+                (
+                    bucket
+                        .resolve(client, namespace.unwrap().as_str())
+                        .await
+                        .context(S3BucketSnafu)
+                        .ok(),
+                    prefix.clone(),
+                )
+            } else {
+                // !!!!!
+                // Ugliness alert!
+                // No point in trying to resolve the connection anymore since there is no
+                // log_file_dir in the first place.
+                // This can casually happen for Spark applications that don't use a history server
+                // !!!!!
+                return Ok(None);
+            };
+
+        // Check that a bucket name has been defined
+        s3bucket
+            .as_ref()
+            .and_then(|i| i.bucket_name.as_ref())
+            .context(BucketNameMissingSnafu)?;
+
+        if let Some(conn) = s3bucket.as_ref().and_then(|i| i.connection.as_ref()) {
+            if let Some(tls) = &conn.tls {
+                match &tls.verification {
+                    TlsVerification::None {} => return S3TlsNoVerificationNotSupportedSnafu.fail(),
+                    TlsVerification::Server(server_verification) => {
+                        match &server_verification.ca_cert {
+                            CaCert::WebPki {} => {}
+                            CaCert::SecretClass(_) => {
+                                return S3TlsCaVerificationNotSupportedSnafu.fail()
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        if let Some(conn) = s3bucket.as_ref().and_then(|i| i.connection.as_ref()) {
+            if conn.tls.as_ref().is_some() {
+                tracing::warn!("The resource indicates S3-access should use TLS: TLS-verification has not yet been implemented \
+            but an HTTPS-endpoint will be used!");
+            }
+        }
```

Comment:
I think this statement is not needed. All the possible cases are handled in the match statement above and are either executed correctly or an error is returned

Reply:
Done.